### PR TITLE
docs: Contour 1.0 release notes

### LINKED ISSUE #1817: Write release notes for 1.0
Write up release notes summarising the changes since 0.15 including -beta.1, rc.1 and rc.2

### PR DESCRIPTION
**Pull request by davecheney**
*** DO NOT MERGE ***

Fixes #1817

This is a first cut of combining the beta.1, rc.1 and rc.2 release
notes. It was mostly pulled together to help me understand what we had
added since 0.15 for the 1.0 announcement blog post.

Signed-off-by: Dave Cheney <dave@cheney.net>

### CODE REVIEW

File: docs/1.0-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+- `make help` target added. Thanks @jpeach.
+- `prefix` conditions must start with a slash. Fixes #1628. Thanks @youngnick.
+- Duplicate HTTPProxy `header` conditions are now rejected. Fixes #1559. Thanks @youngnick.
+- HTTPProxy `route` or `include` blocks with more than one `prefix` condition are now rejected. Fixes #1611. Thanks @stevesloka.
+- The `X-Request-Id` header is now no longer removed from incoming requests. Fixes #1487.
+- `HTTPProxy` `include`s no longer require a `namespace` key. If no `namespace` is provided, the included HTTPProxy is inferred to be in the same namespace as its parent. Fixes #1574. Thanks @youngnick.
+- The ability to write the bootstrap configuration to standard out via `contour bootstrap -- -` has been added. Thanks @jpeach.
+- Contour now validates that TLS certificates either bare the type `kubernetes.io/tls` or, in the case of upstream validation certificates, contain a non empty `ca.crt` key. Fixes #1697. Thanks @jpeach.
+- `x_trace_id` has been added to the set of JSON loggable fields. Fixes #1734. Thanks @cw-sakamoto!
+- Obsolute Heptio branding has been removed from `contour cli`. Thanks @jpeach.
```
Comment (jpeach) [inline docs/1.0-release-notes.md]: s/Obsolute/Obselete/

File: docs/1.0-release-notes.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+- Contour is built with Go 1.13.3.
+
+## Bug fixes
+
+### TLS certificate validation improvements
+
+Contour 1.0.0-rc.2 improves the TLS certificate validation added in rc.1. Contour is now less likely to reject valid certificates that contain unexpected elliptic curve parameters.
```
Comment (jpeach) [inline docs/1.0-release-notes.md]: Should this reference the 1.0.0 release rather than the rc?

File: docs/1.0-release-notes.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+### Contour now waits for a full cache.
+
+Contour now delays serving traffic to Envoy until each of the API informers caught up to the API server. This changes reduces the likelyhood that Envoy can connect to a Contour instance in the process of startup and thus observe an incomplete view of the cluster.
+
+Updates #1280. Thanks @jpeach and @stevesloka.
+
+### HTTPProxy CRD v1
+
+Contour 1.0.0-rc.1 promotes the HTTPProxy CRD to v1. HTTPProxy is now considered stable and our sincere hope is that with the move v1 any changes to the CRD in the future can be made in a backwards compatible manner.
```
Comment (jpeach) [inline docs/1.0-release-notes.md]: Reference 1.0.0 rather than the rc?

File: docs/1.0-release-notes.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+Updates #1280. Thanks @jpeach and @stevesloka.
+
+### HTTPProxy CRD v1
+
+Contour 1.0.0-rc.1 promotes the HTTPProxy CRD to v1. HTTPProxy is now considered stable and our sincere hope is that with the move v1 any changes to the CRD in the future can be made in a backwards compatible manner.  
+
+The move from alpha1 to v1 has resulted in changes to per service health checking, load balancing strategy, and per route prefix rewriting.
+
+Please see the [upgrading document](/docs/upgrading.md) and [HTTPProxy documentation](/docs/httpproxy.md) for advice on upgrading HTTPProxy alpha1 CRDs to v1.
```
Comment (jpeach) [inline docs/1.0-release-notes.md]: IIUC, these won't link if this is being read from GitHub.

File: docs/1.0-release-notes.md (hunk 3 above)
Comment (davecheney) [reply docs/1.0-release-notes.md]: I think there are three mentions of httpproxy in this doc ('cos I just concatenated all the three release notes), I'll dedup and double check.

File: docs/1.0-release-notes.md (hunk 4 above)
Comment (davecheney) [reply docs/1.0-release-notes.md]: oops, these are pre the move of the documentation to netlify. Fixed.

File: docs/1.0-release-notes.md (hunk 2 above)
Comment (davecheney) [reply docs/1.0-release-notes.md]: I dropped this section as it refers to something that happened during the rc process.

**Comment by davecheney** [comment]
Closed.

**Pull request closed without merging**